client/app/components/parameters.js

```diff
@@ -8,13 +8,45 @@ const ParameterSettingsComponent = {
     close: '&',
     dismiss: '&',
   },
-  controller() {
+  controller($sce, Query) {
     'ngInject';
 
+    this.trustAsHtml = html => $sce.trustAsHtml(html);
     this.parameter = this.resolve.parameter;
+    this.onQuerySelect = (query) => {
+      this.parameter.query = query;
```

Comment:
First of all, thanks for review. I am totally new to frontend stuff, and this seemed like something we needed and hence gave it a shot. My knowledge of angular is as much as w3school, stackoverflow and reading redash code.

Regarding the query.id, I was assuming query.id would be sufficient, but I was not sure if the watch on paramter would trigger when the query changes.